lib/usd/ui/layerEditor/mayaSessionState.cpp

```diff
@@ -305,12 +305,27 @@ void MayaSessionState::setupCreateMenu(QMenu* in_menu)
         /*undo*/ false);
 }
 
+const char* getCurrentSaveAsFolderScript = R"(
+global proc string MayaSessionState_GetCurrentSaveAsFolder()
+{
+    string $sceneFolder = dirname(`file -q -sceneName`);
+    if ("" == $sceneFolder)
+    {
+        string $workspaceLocation = `workspace -q -fn`;
+        string $scenesFolder = `workspace -q -fileRuleEntry "scene"`;
+        $sceneFolder = $workspaceLocation + "/" + $scenesFolder;
```

Comment:
First check if there is a current scene, but if not then query the workspace for a good starting folder.  Without this check the dialog will typically open in the current working directory, which could be wherever the Maya executable is.